sources/directconnect/virtual-interface_test.go

```diff
@@ -0,0 +1,101 @@
+package directconnect
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/service/directconnect"
+	"github.com/aws/aws-sdk-go-v2/service/directconnect/types"
+	"github.com/overmindtech/aws-source/sources"
+	"github.com/overmindtech/sdp-go"
+)
+
+func TestVirtualInterfaceOutputMapper(t *testing.T) {
+	output := &directconnect.DescribeVirtualInterfacesOutput{
+		VirtualInterfaces: []types.VirtualInterface{
+			{
+				VirtualInterfaceId:     sources.PtrString("dxvif-ffhhk74f"),
+				ConnectionId:           sources.PtrString("dxcon-fguhmqlc"),
+				VirtualInterfaceState:  "verifying",
+				CustomerAddress:        sources.PtrString("192.168.1.2/30"),
+				AmazonAddress:          sources.PtrString("192.168.1.1/30"),
+				VirtualInterfaceType:   sources.PtrString("private"),
+				VirtualInterfaceName:   sources.PtrString("PrivateVirtualInterface"),
+				DirectConnectGatewayId: sources.PtrString("cf68415c-f4ae-48f2-87a7-3b52cexample"),
+			},
```

Comment:
This has a field called [`VirtualGatewayId`](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/directconnect/types#VirtualInterface) which will refer to https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/directconnect#Client.DescribeVirtualGateways, a type that we will soon create. It's easier to create these links now than to go back and do them later. Since we have the naming convention and we know the CLI command for it: [`describe-virtual-gateways`](https://docs.aws.amazon.com/cli/latest/reference/directconnect/describe-virtual-gateways.html) we can know that the type will be called `directconnect-virtual-gateway` so we can create a link in advance.

Personally I usually try to create test data for every field in the possible return value so it's easy to track which ones I've considered for possible links and which I haven't

Reply:
I think virtaul interface and the virtual gateway are not directly connected. They are indirectly connected over direct connect gateway. That's why I didn't include it in the links on the virtual interface.
```
Each VPC has a virtual private gateway that connects to the Direct Connect gateway using a virtual private gateway association. The Direct Connect gateway uses a private virtual interface for the connection to the AWS Direct Connect location. There is an AWS Direct Connect connection from the location to the customer data center.
```
https://docs.aws.amazon.com/directconnect/latest/UserGuide/direct-connect-gateways-intro.html

What do you think?

Reply:
Yep that reasoning makes sense. In future I think it'd be worth adding `VirtualGatewayId` into the test data, and a comment explaining that reasoning so I know that you've considered it and chosen not to include it rather than potentially missed it